client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/presentation/SessionView.java

```diff
@@ -281,8 +288,18 @@ public boolean onSingleTapUp(MotionEvent e) {
             // send single click
             MotionEvent mappedEvent = mapTouchEvent(e);
             sessionViewListener.onSessionViewBeginTouch();
-            sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), true);
-            sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), false);
+            switch (e.getButtonState()) {
+                case MotionEvent.BUTTON_PRIMARY:
+                    sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), true);
+                    sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), false);
+                    break;
+                case MotionEvent.BUTTON_SECONDARY:
+                    LibFreeRDP.sendCursorEvent(currentSession.getInstance(),
```

Comment:
Maybe a good idea to handle this with the listener just like the primary button?

Reply:
Yes, it can be done. But it will needed 4 lines (right touch and left touch):
                    sessionViewListener.onSessionViewRightTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), true);
                    sessionViewListener.onSessionViewRightTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), false);
                    sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), true);
                    sessionViewListener.onSessionViewLeftTouch((int) mappedEvent.getX(), (int) mappedEvent.getY(), false);